package proto_ff;

// calm 自定义的fieldoption都在该文件中
import "yd_fieldoptions.proto";
import "common_logic.proto";
import "ComDefine.proto";

message PingRsp
{
  required uint64 currentmsTime =1;  //当前的本地时间
  optional uint32 queuereducenum	=2; //排队减少的人数
}

//返回排队结果 SERVER_TO_CLIENT_QUEUE_RESULT
message ServerToClientQueue_RSP
{
  required ERetCode retCode = 1;        //返回错误码
  optional uint32   nNum    =2;		 //当前排队人数
}

// 下线通知
//LOGIC_TO_CLIENT_LOGOUT_LOGICSERVER
message NotifyLogoutGame
{
  message SwitchServerParam
  {
    required string logic_server_ip    = 1;      // 目标logic server的ip
    required uint32 logic_server_port  = 2;      // 目标logic server的port
    required uint32 scene_template_id = 3;       // 跨服目标场景，模板ID就可以
  }

  required LOGOUT_FLAG flag               = 1;
  optional SwitchServerParam switch_server_param = 2; // 当flag==LOGOUT_FLAG_SWITCH_SERVER时候有效
};

//客户端登陆请求(连上网关第一个请求)
//CENTER_TO_CLIENT_LOGIN
message ClientLoginGateRsp
{
  required uint32   ret = 1;
  repeated CharacterDBSimpleInfo charInfoList = 2;
}